### PR TITLE
Add non-deprecated unarchive method

### PR DESCRIPTION
**Pull request by Nadohs**
Adds availability check to either use
new api to unarchive data, or fallback
to deprecated ones on older ios/mac
versions.

addresses issue:
https://github.com/linkedin/cassette/issues/17

### CODE REVIEW

**Comment by nikhilbedi** [comment]
Have we tested that both branches work as expected?

**Comment by Nadohs** [comment]
the existing tests check for archiving and unarchiving working properly for iOS 11+ and macOS 13+, which is the majority of use-cases anyways.
The `else` case here defaults to the usage we had before, on older OS versions before the above replacement API was introduced...
I've manually tested the `else` path, but there are no tests for it.

File: Cassette/CASFileObjectQueue.m [hunk 1]
```diff
@@ -75,4 +75,25 @@ - (void)clear {
...
+
+    if (@available(iOS 11.0, macOS 10.13, *)) {
+        NSError *error;
+        NSKeyedUnarchiver *unarchiver = [[NSKeyedUnarchiver alloc]
+                                         initForReadingFromData:data
+                                         error:&error];
+        [unarchiver setRequiresSecureCoding:NO];
+        result = [unarchiver decodeObjectForKey:NSKeyedArchiveRootObjectKey];
+        if (error != nil) {
+            NSAssert(NO, @"error unarchiving data: %@", error.localizedDescription);
```
Comment (nikhilbedi) [inline Cassette/CASFileObjectQueue.m]: Will this crash the app?

Comment (Nadohs) [reply Cassette/CASFileObjectQueue.m]: the way NSAsserts work is that in debug mode it will stop execution, but in release mode it does not

Comment (Nadohs) [reply Cassette/CASFileObjectQueue.m]: ok i changed it to use `CASLOG`

File: Cassette/CASFileObjectQueue.m [hunk 2]
```diff
@@ -75,4 +75,25 @@ - (void)clear {
     [self.queueFile clear];
 }
 
+#pragma mark - Helper Method
+
+- (id)unarchiveData:(NSData *)data {
```
[the remark this answers is not in the document]
Comment (Nadohs) [reply Cassette/CASFileObjectQueue.m]: changed

**Pull request merged**